# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1523276552 14400
#      Mon Apr 09 08:22:32 2018 -0400
# Node ID ff3d24a818a1c7b04be371d59e4a67e9def56f62
# Parent  512b686110975722fd86e74ecf21231d1f534557
# Parent  7b1b504c2f12d61c4d7e418db962e6e1477f7cb3
maint: Merge stable to default.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -33,17 +33,17 @@ Summary of important user-visible change
 
       glpk  Qhull
 
  ** The octave command no longer starts the GUI by default.  Most users
     starting Octave from command line were expecting the command line
     interface and Desktop launchers required the `--force-gui' option
     anyway.  With this change, desktop launchers should change to use
     the new option `--gui'.  The previous `--force-gui' will continue to
-    work and maps to `--gui' but will be removed in Octave 4.8.
+    work and maps to `--gui' but will be removed in Octave 6.
 
  ** A known bug in Qt (https://bugreports.qt.io/browse/QTBUG-55357) is
     addressed by limiting GUI sub-panel relocation capabilities for Qt
     versions in the range >= 5.6.1 and < 5.7.1.  However, this may not
     thoroughly avoid issues on all platforms.
 
  ** A new container data type--containers.Map--is available.  Map is a
     key/value storage container (a.k.a, a hash) that efficiently allows
@@ -82,17 +82,17 @@ Summary of important user-visible change
 
       quadcc (f, a, b, [AbsTol, [RelTol]])
 
     To update existing code, change instances of RelTol to [0, RelTol].
 
       quadcc (f, a, b, tol) => quadcc (f, a, b, [0, tol])
 
     A warning that a single tolerance input is now interpreted as an
-    absolute tolerance will be issued in Octave versions 4.4 and 4.6,
+    absolute tolerance will be issued in Octave versions 4.4 and 5,
     after which it will be removed.  The warning has ID
     "Octave:quadcc:RelTol-conversion" and can be disabled with
 
       warning ("off", "Octave:quadcc:RelTol-conversion")
 
  ** The qr function now returns a standard factorization unless
     explicitly instructed to perform an economy factorization by using a
     final argument of 0.
@@ -314,17 +314,17 @@ Summary of important user-visible change
       yticklabels
       yticks
       zticklabels
       zticks
 
  ** Deprecated functions.
 
     The following functions have been deprecated in Octave 4.4 and will
-    be removed from Octave 4.8 (or whatever version is the second major
+    be removed from Octave 6 (or whatever version is the second major
     release after 4.4):
 
       Function             | Replacement
       ---------------------|------------------
       chop                 | sprintf for visual results
       desktop              | isguirunning
       tmpnam               | tempname
       toascii              | double
@@ -357,17 +357,17 @@ Summary of important user-visible change
 
  ** The "Octave:undefined-return-values" warning ID is obsolete.  Octave
     now throws an error for any attempts to assign undefined values that
     might be returned from functions.
 
  ** Deprecated graphics properties.
 
     The following properties or allowed corresponding values have been
-    deprecated in Octave 4.4 and will be removed from Octave 4.8 (or whatever
+    deprecated in Octave 4.4 and will be removed from Octave 6 (or whatever
     version is the second major release after 4.4):
 
       Object               | Property                | Value
       ---------------------|-------------------------|-------------------
       figure               | doublebuffer            |
                            | mincolormap             |
                            | wvisual                 |
                            | wvisualmode             |
@@ -377,17 +377,17 @@ Summary of important user-visible change
       axes                 | drawmode                |
       annotation           | edgecolor ("rectangle") |
       text                 | fontweight              | "demi" and "light"
       uicontrol            | fontweight              | "demi" and "light"
       uipanel              | fontweight              | "demi" and "light"
       uibuttongroup        | fontweight              | "demi" and "light"
 
  ** The rectangle and ellipse annotation property "edgecolor" has been
-    deprecated and will be removed from Octave 4.8 (or whatever version
+    deprecated and will be removed from Octave 6 (or whatever version
     is the second major release after 4.4).  Use the property "color"
     instead.
 
  ** The header file oct-alloc.h has been removed along with the macros
     that it defined (DECLARE_OCTAVE_ALLOCATOR, DEFINE_OCTAVE_ALLOCATOR,
     and DEFINE_OCTAVE_ALLOCATOR2).
 
 
@@ -901,17 +901,17 @@ Summary of important user-visible change
     eventually be removed.  This property has also been removed
     from Matlab, and was never implemented in Octave.
 
  ** The graphics property "graphicssmoothing" for figures now controls
     whether anti-aliasing will be used for lines.  The default is "on".
 
  ** The value "zero" for the axes properties "xaxislocation" and
     "yaxislocation" has been deprecated and will be removed from
-    Octave 4.6.  Use "origin" instead.
+    Octave 5.  Use "origin" instead.
 
  ** The publish function allows easy publication of Octave script files
     in HTML or other formats, including figures and output created by
     this script.  It comes with its counterpart grabcode, which lets one
     literally grab the HTML published code from a remote website, for
     example.
 
  ** The value of the MEX variable TrapFlag now defaults to 0, which will
@@ -962,17 +962,17 @@ Summary of important user-visible change
       reducepatch
       reducevolume
       smooth3
       uibuttongroup
 
  ** Deprecated functions.
 
     The following functions have been deprecated in Octave 4.2 and will
-    be removed from Octave 4.6 (or whatever version is the second major
+    be removed from Octave 5 (or whatever version is the second major
     release after 4.2):
 
       Function             | Replacement
       ---------------------|------------------
       bitmax               | flintmax
       mahalanobis          | mahal in Octave-Forge statistics pkg
       md5sum               | hash
       octave_config_info   | __octave_config_info__
@@ -1008,17 +1008,17 @@ Summary of important user-visible change
     "current_liboctave_error_with_id_handler".  If a programmer has
     installed their own custom error handling routines when directly
     linking with liboctave then these must be updated to throw an
     exception and not return to the calling program.
 
  ** The system for common errors and warnings has been renamed from
     gripe_XXX to either err_XXX if error is called or warn_XXX if
     warning is called.  The gripe_XXX functions are deprecated and will
-    be removed in version 4.6.
+    be removed in version 5.
 
  ** New configure option, --enable-address-sanitizer-flags, to build
     Octave with memory allocator checks (similar to those in valgrind)
     built in.
 
 Summary of important user-visible changes for version 4.0 (2015-05-23):
 ----------------------------------------------------------------------
 
diff --git a/doc/interpreter/bugs.txi b/doc/interpreter/bugs.txi
--- a/doc/interpreter/bugs.txi
+++ b/doc/interpreter/bugs.txi
@@ -25,17 +25,17 @@
 @end ifclear
 
 @ifset BUGSONLY
 @include macros.texi
 
 This file documents known bugs in Octave and describes where and how to
 report any bugs that you may find.
 
-Copyright (C) 1996, 1997, 2007 John W. Eaton.  You may copy, distribute, and
+Copyright (C) 1996-2018 John W. Eaton.  You may copy, distribute, and
 modify it freely as long as you preserve this copyright notice and
 permission notice.
 
 @node Trouble
 @chapter Known Causes of Trouble with Octave
 @end ifset
 
 @cindex bugs, known
diff --git a/libgui/src/octave-dock-widget.cc b/libgui/src/octave-dock-widget.cc
--- a/libgui/src/octave-dock-widget.cc
+++ b/libgui/src/octave-dock-widget.cc
@@ -455,17 +455,17 @@ namespace octave
       m_predecessor_widget->focus ();
 
     m_predecessor_widget = nullptr;
     // FIXME: Until cset bda0c5b38bda, the wrong keys "Dockwidget/..." were used
     // here.  This had no effect in Qt4, but does in Qt5.  In the following, the
     // four incorrect keys are updated if still present in the settings files.
     // The keys are also used in the settings dialog, but
     // octave_dock_widget::handle_settings is already called at program start.
-    // These tests can be removed in a future version of Octave (version 4.8).
+    // These tests can be removed in a future version of Octave (version 6).
     resource_manager::update_settings_key ("Dockwidgets/title_bg_color",
                                            "DockWidgets/title_bg_color");
     resource_manager::update_settings_key ("Dockwidgets/title_bg_color_active",
                                            "DockWidgets/title_bg_color_active");
     resource_manager::update_settings_key ("Dockwidgets/title_fg_color",
                                            "DockWidgets/title_fg_color");
     resource_manager::update_settings_key ("Dockwidgets/title_fg_color_active",
                                            "DockWidgets/title_fg_color_active");
diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -3176,17 +3176,17 @@ public:
       bool_property __printing__ h , "off"
       any_property __pan_mode__ h , Matrix ()
       any_property __plot_stream__ h , Matrix ()
       any_property __rotate_mode__ h , Matrix ()
       any_property __zoom_mode__ h , Matrix ()
 
       // Obsolete properties: doublebuffer, mincolormap, wvisual, wvisualmode,
       //                      xdisplay, xvisual, xvisualmode
-      // FIXME: Remove in version 4.6
+      // FIXME: DEPRECATED: Remove in version 5.
       bool_property doublebuffer hd , "on"
       double_property mincolormap hd , 64
       string_property wvisual hmd , ""
       radio_property wvisualmode hd , "{auto}|manual"
       string_property xdisplay hd , ""
       string_property xvisual hmd , ""
       radio_property xvisualmode hd , "{auto}|manual"
     END_PROPERTIES
@@ -4283,19 +4283,20 @@ public:
   public:
     // See the genprops.awk script for an explanation of the
     // properties declarations.
     // Programming note: Keep property list sorted if new ones are added.
 
     BEGIN_PROPERTIES (line)
       color_property color , color_property (color_values (0, 0, 0), radio_values ("none"))
       string_property displayname , ""
-      // FIXME: Remove erasemode property in version 4.6.  (rm all instances in file)
+      // FIXME: DEPRECATED: Remove erasemode property in version 5
+      // (rm all instances in file).
       radio_property erasemode h , "{normal}|none|xor|background"
-      // FIXME: Remove interpreter property in version 4.8
+      // FIXME: DEPRECATED: Remove interpreter property in version 6.
       radio_property interpreter hd , "{tex}|none|latex"
       radio_property linejoin , "{round}|miter|chamfer"
       radio_property linestyle , "{-}|--|:|-.|none"
       double_property linewidth , 0.5
       radio_property marker , "{none}|+|o|*|.|x|s|square|d|diamond|^|v|>|<|p|pentagram|h|hexagram"
       color_property markeredgecolor , color_property (radio_values ("{auto}|none"), color_values (0, 0, 0))
       color_property markerfacecolor , color_property (radio_values ("auto|{none}"), color_values (0, 0, 0))
       double_property markersize , 6
@@ -4496,17 +4497,18 @@ public:
     void update_fontname (void) { update_font (); update_text_extent (); }
     void update_fontsize (void) { update_font (); update_text_extent (); }
     void update_fontangle (void) { update_font (); update_text_extent (); }
 
     void update_fontweight (void)
     {
       update_font ();
       update_text_extent ();
-      // FIXME: Remove warning with demi and light in 4.7+
+      // FIXME: DEPRECATED: Remove warning with demi and light in
+      // version 6.
       if (fontweight.is ("demi") || fontweight.is ("light"))
         warning_with_id ("Octave:deprecated-property",
                          "Setting 'fontweight' to '%s' is deprecated, \
 use 'normal' or 'bold'.", fontweight.current_value ().c_str ());
     }
 
     void update_interpreter (void) { update_text_extent (); }
     void update_horizontalalignment (void) { update_text_extent (); }
@@ -4852,17 +4854,17 @@ public:
       double_radio_property facealpha , double_radio_property (1.0, radio_values ("flat|interp"))
       color_property facecolor , color_property (color_values (0, 0, 0), radio_values ("none|flat|interp"))
       radio_property facelighting , "none|{flat}|gouraud|phong"
       array_property facenormals m , Matrix ()
       radio_property facenormalsmode , "{auto}|manual"
       array_property faces u , default_patch_faces ()
       array_property facevertexalphadata , Matrix ()
       array_property facevertexcdata u , Matrix ()
-      // FIXME: Remove interpreter property in version 4.8
+      // FIXME: DEPRECATED: Remove interpreter property in version 6.
       radio_property interpreter hd , "{tex}|none|latex"
       radio_property linestyle , "{-}|--|:|-.|none"
       double_property linewidth , 0.5
       radio_property marker , "{none}|+|o|*|.|x|s|square|d|diamond|^|v|>|<|p|pentagram|h|hexagram"
       color_property markeredgecolor , color_property (radio_values ("none|{auto}|flat"), color_values (0, 0, 0))
       color_property markerfacecolor , color_property (radio_values ("{none}|auto|flat"), color_values (0, 0, 0))
       double_property markersize , 6
       double_property specularcolorreflectance , 1.0
@@ -5046,17 +5048,17 @@ public:
       color_property edgecolor , color_property (color_values (0, 0, 0), radio_values ("none|flat|interp"))
       radio_property edgelighting , "{none}|flat|gouraud|phong"
       radio_property erasemode h , "{normal}|none|xor|background"
       double_radio_property facealpha , double_radio_property (1.0, radio_values ("flat|interp|texturemap"))
       color_property facecolor , color_property (radio_values ("none|{flat}|interp|texturemap"), color_values (0, 0, 0))
       radio_property facelighting , "none|{flat}|gouraud|phong"
       array_property facenormals m , Matrix ()
       radio_property facenormalsmode , "{auto}|manual"
-      // FIXME: Remove interpreter property in version 4.8
+      // FIXME: DEPRECATED: Remove interpreter property in version 6.
       radio_property interpreter hd , "{tex}|none|latex"
       radio_property linestyle , "{-}|--|:|-.|none"
       double_property linewidth , 0.5
       radio_property marker , "{none}|+|o|*|.|x|s|square|d|diamond|^|v|>|<|p|pentagram|h|hexagram"
       color_property markeredgecolor , color_property (radio_values ("none|{auto}|flat"), color_values (0, 0, 0))
       color_property markerfacecolor , color_property (radio_values ("{none}|auto|flat"), color_values (0, 0, 0))
       double_property markersize , 6
       radio_property meshstyle , "{both}|row|column"
@@ -5472,17 +5474,18 @@ public:
     void update_string (void) { update_text_extent (); }
     void update_fontname (void) { update_text_extent (); }
     void update_fontsize (void) { update_text_extent (); }
     void update_fontangle (void) { update_text_extent (); }
 
     void update_fontweight (void)
     {
       update_text_extent ();
-      // FIXME: Remove warning with demi and light in 4.7+
+      // FIXME: DEPRECATED: Remove warning with demi and light in
+      // version 6.
       if (fontweight.is ("demi") || fontweight.is ("light"))
         warning_with_id ("Octave:deprecated-property",
                          "Setting 'fontweight' to '%s' is deprecated, \
 use 'normal' or 'bold'.", fontweight.current_value ().c_str ());
     }
 
     void update_fontunits (const caseless_str& old_units);
 
@@ -5569,17 +5572,17 @@ public:
     // void update_text_extent (void);
     // void update_string (void) { update_text_extent (); }
     // void update_fontname (void) { update_text_extent (); }
     // void update_fontsize (void) { update_text_extent (); }
     // void update_fontangle (void) { update_text_extent (); }
 
     void update_fontweight (void)
     {
-      // FIXME: Remove this warning in 4.7+
+      // FIXME: DEPRECATED: Remove this warning in version 6.
       if (fontweight.is ("demi") || fontweight.is ("light"))
         warning_with_id ("Octave:deprecated-property",
                          "Setting 'fontweight' to '%s' is deprecated, \
 use 'normal' or 'bold'.", fontweight.current_value ().c_str ());
     }
 
     void update_units (const caseless_str& old_units);
     void update_fontunits (const caseless_str& old_units);
@@ -5658,17 +5661,17 @@ public:
       position.add_constraint (dim_vector (1, 4));
     }
 
     void update_units (const caseless_str& old_units);
     void update_fontunits (const caseless_str& old_units);
 
     void update_fontweight (void)
     {
-      // FIXME: Remove this warning in 4.7+
+      // FIXME: DEPRECATED: Remove this warning in version 6.
       if (fontweight.is ("demi") || fontweight.is ("light"))
         warning_with_id ("Octave:deprecated-property",
                          "Setting 'fontweight' to '%s' is deprecated, \
 use 'normal' or 'bold'.", fontweight.current_value ().c_str ());
     }
   };
 
 private:
diff --git a/libinterp/corefcn/oct-obj.h b/libinterp/corefcn/oct-obj.h
--- a/libinterp/corefcn/oct-obj.h
+++ b/libinterp/corefcn/oct-obj.h
@@ -17,17 +17,17 @@ GNU General Public License for more deta
 You should have received a copy of the GNU General Public License
 along with Octave; see the file COPYING.  If not, see
 <https://www.gnu.org/licenses/>.
 
 */
 
 ////////////////////////////////////////////////////////////////////////
 // Note: oct-obj.h was deprecated in version 4.2 and will be removed in
-// version 4.6.
+// version 5.
 ////////////////////////////////////////////////////////////////////////
 
 #if ! defined (octave_oct_obj_h)
 #define octave_oct_obj_h 1
 
 #include "octave-config.h"
 
 #warning oct-obj.h has been deprecated; use ovl.h instead
diff --git a/libinterp/corefcn/quadcc.cc b/libinterp/corefcn/quadcc.cc
--- a/libinterp/corefcn/quadcc.cc
+++ b/libinterp/corefcn/quadcc.cc
@@ -1546,20 +1546,20 @@ other methods such as @code{quadgk}.
 
 The algorithm uses @nospell{Clenshaw-Curtis} quadrature rules of increasing
 degree in each interval and bisects the interval if either the function does
 not appear to be smooth or a rule of maximum degree has been reached.  The
 error estimate is computed from the L2-norm of the difference between two
 successive interpolations of the integrand over the nodes of the respective
 quadrature rules.
 
-@c FIXME: Remove in Octave version 4.8
+@c FIXME: DEPRECATED: Remove in Octave version 6.
 Implementation Note: For Octave versions @leq{} 4.2, @code{quadcc} accepted a
 single tolerance argument which specified the relative tolerance.  For
-versions 4.4 and 4.6, @code{quadcc} will issue a warning when called with a
+versions 4.4 and 5, @code{quadcc} will issue a warning when called with a
 single tolerance argument indicating that the meaning of this input has
 changed from relative tolerance to absolute tolerance.  The warning ID for this
 message is @qcode{"Octave:quadcc:RelTol-conversion"}.  The warning may be
 disabled with @code{warning ("off", "Octave:quadcc:RelTol-conversion")}.
 
 Reference: @nospell{P. Gonnet}, @cite{Increasing the Reliability of Adaptive
 Quadrature Using Explicit Interpolants}, @nospell{ACM} Transactions on
 Mathematical Software, Vol. 37, Issue 3, Article No. 3, 2010.
@@ -1642,17 +1642,17 @@ Mathematical Software, Vol. 37, Issue 3,
       NDArray tol = args(3).array_value ();
 
       abstol = tol(0);
       if (abstol < 0)
         error ("quadcc: absolute tolerance must be >=0");
 
       if (tol.numel () == 1)
         {
-          // FIXME: Remove warning in Octave version 4.8
+          // FIXME: DEPRECATED: Remove warning in Octave version 6.
           static bool do_warn = true;
           if (do_warn)
             {
               warning_with_id ("Octave:quadcc:RelTol-conversion",
                                "A single tolerance input now sets AbsTol, not RelTol");
               do_warn = false;
             }
 
diff --git a/libinterp/options-usage.h b/libinterp/options-usage.h
--- a/libinterp/options-usage.h
+++ b/libinterp/options-usage.h
@@ -54,17 +54,18 @@ static const char *short_opts = "+HWVdfh
 #endif
 
 // Long options.  See the comments in getopt.h for the meanings of the
 // fields in this structure.
 #define BUILT_IN_DOCSTRINGS_FILE_OPTION 1
 #define DOC_CACHE_FILE_OPTION 2
 #define EVAL_OPTION 3
 #define EXEC_PATH_OPTION 4
-#define FORCE_GUI_OPTION 5 // ignored since Octave 4.4, remove for 4.8
+// FIXME: DEPRECATED: ignored since Octave 4.4, remove for version 6.
+#define FORCE_GUI_OPTION 5
 #define GUI_OPTION 5
 #define IMAGE_PATH_OPTION 6
 #define INFO_FILE_OPTION 7
 #define INFO_PROG_OPTION 8
 #define DEBUG_JIT_OPTION 9
 #define JIT_COMPILER_OPTION 10
 #define LINE_EDITING_OPTION 11
 #define NO_GUI_OPTION 12
diff --git a/scripts/deprecated/tmpnam.m b/scripts/deprecated/tmpnam.m
--- a/scripts/deprecated/tmpnam.m
+++ b/scripts/deprecated/tmpnam.m
@@ -32,17 +32,17 @@
 ## otherwise the default directory for temporary files is used.
 ##
 ## Programming Note: Because the named file is not opened by @code{tmpnam},
 ## it is possible, though relatively unlikely, that it will not be available
 ## by the time your program attempts to open it.  If this is a concern,
 ## see @code{tmpfile}.  The functions @code{tmpnam} and @code{tempname} are
 ## equivalent with the latter provided for @sc{matlab} compatibility.
 ##
-## @strong{Caution}: @code{tmpnam} will be removed in Octave version 4.8, or
+## @strong{Caution}: @code{tmpnam} will be removed in Octave version 6, or
 ## whatever the version number is two releases from 4.4. Use the equivalent
 ## @code{tempname} in all new code.
 ## @seealso{tempname, mkstemp, tempdir, P_tmpdir, tmpfile}
 ## @end deftypefn
 
 function filename = tmpnam (varargin)
 
   persistent warned = false;
diff --git a/scripts/deprecated/toascii.m b/scripts/deprecated/toascii.m
--- a/scripts/deprecated/toascii.m
+++ b/scripts/deprecated/toascii.m
@@ -31,17 +31,17 @@
 ## toascii ("ASCII"
 ##      @result{} [ 65, 83, 67, 73, 73 ]
 ## @end group
 ##
 ## @end example
 ## @seealso{double, char}
 ## @end deftypefn
 
-## Remove in version 4.8.
+## FIXME: DEPRECATED: Remove in version 6.
 
 function retval = toascii (str)
 
   persistent warned = false;
   if (! warned)
     warned = true;
     warning ("Octave:deprecated-function",
              "toascii is obsolete and will be removed from a future version of Octave, please use double instead");
diff --git a/scripts/general/repelem.m b/scripts/general/repelem.m
--- a/scripts/general/repelem.m
+++ b/scripts/general/repelem.m
@@ -1,9 +1,9 @@
-## Copyright (C) 2015 Markus Bergholz
+## Copyright (C) 2015-2018 Markus Bergholz
 ##
 ## This file is part of Octave.
 ##
 ## Octave is free software: you can redistribute it and/or modify it
 ## under the terms of the GNU General Public License as published by
 ## the Free Software Foundation, either version 3 of the License, or
 ## (at your option) any later version.
 ##
diff --git a/scripts/plot/appearance/annotation.m b/scripts/plot/appearance/annotation.m
--- a/scripts/plot/appearance/annotation.m
+++ b/scripts/plot/appearance/annotation.m
@@ -475,17 +475,17 @@ function h = buildannot (hax, objtype, p
       if (strcmp (objtype, "rectangle"))
         [x, y] = pos2rect (pos);
       else
         [x, y] = pos2ell (pos);
       endif
 
       hr = patch (x, y, "parent", h);
 
-      ## FIXME: Remove warn state switching in Octave 4.8
+      ## FIXME: DEPRECATED: Remove warn state switching in version 6.
       old_warn_state = warning ("off", "Octave:deprecated-property");
       propnames = rectprops ("names");
       for ii = 1:numel (propnames)
         update_rect (h, {}, propnames{ii}, hr, objtype);
       endfor
       warning (old_warn_state);
 
       rectmenu (hui, h);
@@ -795,17 +795,17 @@ function textboxmenu (hui, hpar)
   prop = "linewidth";
   vals = 0.5:.5:2;
   addbasemenu (hm2, hpar, prop, vals, "Line Width");
 
 endfunction
 
 function props = rectprops (varargin)
 
-  ## FIXME: Remove "edgecolor" in Octave 4.8
+  ## FIXME: DEPRECATED: Remove "edgecolor" in version 6.
   props = {"color", "patchedgecolor", "k", ...
            "edgecolor", "patchedgecolor", "k", ...
            "facealpha", "patchfacealpha", 1, ...
            "facecolor", "patchfacecolor", "none", ...
            "linestyle", "patchlinestyle", "-", ...
            "linewidth", "patchlinewidth", 0.5};
   if (strcmp (varargin, "names"))
     props = props(1:3:end);
@@ -1274,17 +1274,17 @@ function update_rect (h, ~, prop, hre, t
           [x, y] = pos2ell (pos);
         endif
         set (hre, "xdata", x, "ydata", y);
 
       case "color"
         set (hre, "edgecolor", get (h, prop));
 
       case "edgecolor"
-        ## FIXME: Remove "edgecolor" in Octave 4.8
+        ## FIXME: DEPRECATED: Remove "edgecolor" in version 6.
         warning ("Octave:deprecated-property",
                  ['annotation: Property "edgecolor" for ' typ ' annotations'...
                   ' is deprecated and will be removed from a future version'...
                   ' of Octave.  Use "color" instead.']);
         set (hre, "edgecolor", get (h, prop));
 
       otherwise
         set (hre, prop, get (h, prop));
diff --git a/test/deprecate-props.tst b/test/deprecate-props.tst
--- a/test/deprecate-props.tst
+++ b/test/deprecate-props.tst
@@ -29,54 +29,54 @@
 %!      error ("Please remove %s property %s", get (h, "type"), prop);
 %!    elseif (! isempty (val) && any (strcmp (val, set (h, prop))))
 %!      error ("Please remove '%s' from allowed values for %s property %s",
 %!             val, get (h, "type"), prop);
 %!    endif
 %!  endif
 %!endfunction
 
-## annotation rectangle "edgecolor" deprecated in 4.4, remove from 6.
+## annotation rectangle "edgecolor" deprecated in 4.4, remove from version 6.
 %!test
 %! hf = figure ("visible", "off");
 %! unwind_protect
 %!   ha = annotation ("rectangle");
 %!   testprop (ha, "edgecolor", "6.0");
 %! unwind_protect_cleanup
 %!   close (hf);
 %! end_unwind_protect
 
 ## figure "doublebuffer, mincolormap, wvisual, wvisualmode, xdisplay,
-## xvisual, xvisualmode" deprecated in 4.4, remove from 6.
+## xvisual, xvisualmode" deprecated in 4.4, remove from version 6.
 %!test
 %! hf = figure ("visible", "off");
 %! unwind_protect
 %!   testprop (hf, "doublebuffer", "6.0");
 %!   testprop (hf, "mincolormap", "6.0");
 %!   testprop (hf, "wvisual", "6.0");
 %!   testprop (hf, "wvisualmode", "6.0");
 %!   testprop (hf, "xdisplay", "6.0");
 %!   testprop (hf, "xvisual", "6.0");
 %!   testprop (hf, "xvisualmode", "6.0");
 %! unwind_protect_cleanup
 %!   close (hf)
 %! end_unwind_protect
 
-## axes "drawmode" deprecated in 4.4, remove from 6.
+## axes "drawmode" deprecated in 4.4, remove from version 6.
 %!test
 %! hf = figure ("visible", "off");
 %! unwind_protect
 %!   hax = axes ();
 %!   testprop (hax, "drawmode", "6.0");
 %! unwind_protect_cleanup
 %!   close (hf);
 %! end_unwind_protect
 
 ## text/uicontrol/uipanel/uibuttongroup  "demi" and "light" values for
-## "fontweight" property are deprecated in 4.4, remove from 6.
+## "fontweight" property are deprecated in 4.4, remove from version 6:
 ##   * remove "demi" and "light" options in graphics.in.h,
 ##   QtHandlesUtils.cc and ft-text-renderer.cc
 ##   * remove warnings from update_fontweight in graphics.in.h
 %!test
 %! hf = figure ("visible", "off");
 %! unwind_protect
 %!   ht = text ();
 %!   testprop (ht, "fontweight", "6.0", "demi");
